'panic: runtime error' occurred

4 nodes based on https://github.com/bosnet/sebak/tree/e076904d276c6b5c5d88a1dce6d9f60216e44dfe .

The full logs can be found at https://gist.github.com/spikeekips/37c64a1084307802d8858339563872e5 .
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x921c12]

goroutine 12210 [running]:
boscoin.io/sebak/lib/consensus.(*RunningRound).RoundVote(...)
	/sebak/lib/consensus/running_round.go:40
boscoin.io/sebak/lib/consensus.(*ISAAC).CanGetVotingResult(0xc00017e000, 0x10ff831, 0x1, 0xc0010e6570, 0x2c, 0xc000c99a40, 0x58, 0xc000c997a0, 0x58, 0xc00016dce0, ...)
	/sebak/lib/consensus/isaac.go:214 +0x182
boscoin.io/sebak/lib/node/runner.BallotCheckResult(0xc41e20, 0xc0010e4b40, 0x0, 0x0, 0x0, 0x0, 0x0)
	/sebak/lib/node/runner/checker.go:375 +0xbe
boscoin.io/sebak/lib/common.RunChecker(0xc41e20, 0xc0010e4b40, 0xbaead8, 0x0, 0x0, 0x0, 0x0, 0x0)
	/sebak/lib/common/checker.go:55 +0x102
boscoin.io/sebak/lib/node/runner.(*NodeRunner).handleBallotMessage(0xc000180000, 0xb5bea0, 0xc001031340, 0xc001031340, 0xc000318e60)
	/sebak/lib/node/runner/node_runner.go:538 +0x52c
boscoin.io/sebak/lib/node/runner.(*NodeRunner).broadcastBallot.func1(0xc000180000, 0xc001030c60)
	/sebak/lib/node/runner/node_runner.go:698 +0x91
created by boscoin.io/sebak/lib/node/runner.(*NodeRunner).broadcastBallot
	/sebak/lib/node/runner/node_runner.go:697 +0xa3
exit status 2
```

--- a/lib/consensus/isaac.go
+++ b/lib/consensus/isaac.go
@@ -2,7 +2,6 @@ package consensus
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"sync"
 
@@ -11,6 +10,7 @@ import (
 	"boscoin.io/sebak/lib/ballot"
 	"boscoin.io/sebak/lib/block"
 	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/errors"
 	"boscoin.io/sebak/lib/network"
 	"boscoin.io/sebak/lib/node"
 	"boscoin.io/sebak/lib/storage"
@@ -210,7 +210,14 @@ func (is *ISAAC) Vote(b ballot.Ballot) (isNew bool, err error) {
 func (is *ISAAC) CanGetVotingResult(b ballot.Ballot) (RoundVoteResult, voting.Hole, bool) {
 	is.RLock()
 	defer is.RUnlock()
-	runningRound, _ := is.RunningRounds[b.VotingBasis().Index()]
+
+	is.log.Debug("CanGetVotingResult", "ballot", b)
+	runningRound, found := is.RunningRounds[b.VotingBasis().Index()]
+	if !found {
+		// if RunningRound is not found, this ballot will be stopped.
+		return nil, voting.NOTYET, true
+	}
+
 	if roundVote, err := runningRound.RoundVote(b.Proposer()); err == nil {
 		return roundVote.CanGetVotingResult(is.policy, b.State(), is.log)
 	} else {
@@ -221,7 +228,12 @@ func (is *ISAAC) CanGetVotingResult(b ballot.Ballot) (RoundVoteResult, voting.Ho
 func (is *ISAAC) IsVotedByNode(b ballot.Ballot, node string) (bool, error) {
 	is.RLock()
 	defer is.RUnlock()
-	runningRound, _ := is.RunningRounds[b.VotingBasis().Index()]
+
+	runningRound, found := is.RunningRounds[b.VotingBasis().Index()]
+	if !found {
+		return false, errors.RoundVoteNotFound
+	}
+
 	if roundVote, err := runningRound.RoundVote(b.Proposer()); err == nil {
 		return roundVote.IsVotedByNode(b.State(), node), nil
 	} else {

--- a/lib/node/runner/checker.go
+++ b/lib/node/runner/checker.go
@@ -374,6 +374,10 @@ func BallotCheckResult(c common.Checker, args ...interface{}) (err error) {
 	}
 
 	result, votingHole, finished := checker.NodeRunner.Consensus().CanGetVotingResult(checker.Ballot)
+	if votingHole == voting.NOTYET && finished {
+		err = NewCheckerStopCloseConsensus(checker, "ballot already finished")
+		return
+	}
 
 	checker.Result = result
 	checker.VotingFinished = finished

